onnx/funconnx/op_run.py

```diff
@@ -0,0 +1,508 @@
+            for k, v in self._schema.attributes.items():  # type: ignore
+                if not hasattr(self, k):
+                    if getattr(v, "required", True):
+                        raise RuntimeError(
+                            f"Attribute {k!r} is expected based on ONNX specifications "
+                            f"for node {self.onnx_node.op_type!r}."
+                        )
+                    if hasattr(v, "default_value"):
+                        if v.default_value.type == 0:
+                            # default value is undefined, it depends on the inputs
+                            value = None  # type: ignore
+                        else:
+                            value = self._extract_attribute_value(v.default_value, v)
+                        setattr(self, k, value)
+                        added_attributes.append(k)
+        self.attributes_names_ = set(added_attributes)
+
+    @staticmethod
+    def local_inputs(graph: GraphProto) -> List[str]:
+        """
+        Returns all varibles not registered as inputs and not produced by
+        an node inside the graph. This inputs are part of the context
+        existing in the graph calling this one.
+        """
+        if not isinstance(graph, GraphProto):
+            raise TypeError(f"Unexpected type {type(graph)!r}.")
+        local = set()
+        known = set()
+        for init in graph.initializer:
+            known.add(init.name)
+        for sparse_init in graph.sparse_initializer:
+            known.add(sparse_init.name)  # type: ignore
+        for inp in graph.input:
+            known.add(inp.name)
+        for node in graph.node:
+            for o in node.output:
+                known.add(o)
+            for i in node.input:
+                if i not in known:
+                    local.add(i)
+        return list(local)
+
+    @property
+    def input(self) -> Iterable[str]:
+        "Returns node attribute `input`."
+        return self.onnx_node.input  # type: ignore
+
+    @property
+    def output(self) -> Iterable[str]:
+        "Returns node attribute `output`."
+        return self.onnx_node.output  # type: ignore
+
+    def attr(
+        self, *names: str, overridden_attributes: Optional[Dict[str, Any]] = None
+    ) -> Any:
+        """
+        Retrieves the value of an attribute. It is `self.<name>` unless
+        its value is linked to the function attribute it is part of.
+        """
+        values = []
+        for name in names:
+            value = getattr(self, name)
+            if isinstance(value, RefAttrName):
+                if overridden_attributes is None:
+                    raise AttributeError(
+                        f"Attribute {name!r} of operator {type(self)} is linked but "
+                        f"'overridden_attributes' has no value for it."
+                    )
+                values.append(overridden_attributes[name])
```

Comment:
I think `name` should be replaced by `value.name` here, if my understanding is correct.